### ISSUE TITLE
Use xgboost to categorize a continuous variable, conditional on prediction and other covariates

### ISSUE THREAD

**Issue by bakaburg1** [dmlc/xgboost#4787]
Hello,

I'm doing causal inference using the virtual twins method (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3880775/). The literature on this and similar method focuses on categorical treatments, while in my case I have continuous ones.

To categorize such variables I tested two methods: Either I use a regression tree on the xgboost model predictions to find subsets the maximize the discrimination of the outcome, or, if the tree fails to find useful strata, I categorize the variable using quartiles. 
The first method has the problem of not considering the other covariates, even if is trained on the full xgboost model predictions,  which do include the effect of all variables, not on the actual dichotomic outcome. The second method instead doesn't even consider the outcome.

Since xgboost internally do split the predictors among the trees, is there a theoretically robust method to use such rules to transform the continuous predictors in a categorical one with as few splits as possible which include information about the outcome and the covariates. A sort of simpler rule produced by collapsing all the xgboost tree rules related to the predictor of interest.

Thanks

**Comment by hcho3** [comment]
@avinashbarnwal Are you familiar with the topic being discussed here? Are decision trees commonly used in causal inference?

**Comment by avinashbarnwal** [comment]
@hcho3, i have heard about it. I have seen one R package vignette explaining it more clearly. I think we can see the performance difference between random forest and xgboost. I am not sure about how we can have internal splits based on that. Generally, researchers also Bayesian trees for internal splits.

Link - https://cran.r-project.org/web/packages/aVirtualTwins/vignettes/full-example.html

**Comment by bakaburg1** [comment]
Yes, the context is similar. The problem is that the whole literature on the matter is based on a dichotomic treatment variable, while I need to extend it to continous ones. One method I thought of is to transform the variable in a categorical one using meaningful split points.
